### ISSUE TITLE
./x2t: error while loading shared libraries: libgraphics.so: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by plush** [nextcloud/documentserver_community#195]
Since the upgrade to Nextcloud 20.0 and Community Document Server 0.1.8, any attempt to access a document results in "ONLYOFFICE cannot be reached. Please contact admin." The log says that this is because the `x2t` binary cannot find `libgraphics.so`. The two files actually reside in the same directory, but apparently the library search path was not set, so `x2t` cannot find the library it needs. Setting `LD_LIBRARY_PATH=.` allows the binary to run. You can verify this on the command line.

Without setting `LD_LIBRARY_PATH`: Failure
```
$ cd apps/documentserver_community/3rdparty/onlyoffice/documentserver/server/FileConverter/bin
$ ./x2t
./x2t: error while loading shared libraries: libgraphics.so: cannot open shared object file: No such file or directory
```

With `LD_LIBRARY_PATH=.`: Success
```
$ cd apps/documentserver_community/3rdparty/onlyoffice/documentserver/server/FileConverter/bin
$ LD_LIBRARY_PATH=. ./x2t
-------------------------------------------------------------------------------
                OOX/binary file converter. Version: 5.6.5.3
-------------------------------------------------------------------------------
[...]
```

The fix would be to pass the correct `LD_LIBRARY_PATH` in ConverterBinary.php line 51. I quickly hacked this for myself by setting `LD_LIBRARY_PATH` to `.`, but the proper fix may be to derive the value from a path constant somewhere - I will leave it to those experienced with this project to develop a patch. This one-line hack was sufficient to get my Nextcloud working again though.

**Comment by speerwerfer** [comment]
> The fix would be to pass the correct `LD_LIBRARY_PATH` in ConverterBinary.php line 51. I quickly hacked this for myself by setting `LD_LIBRARY_PATH` to `.`, but the proper fix may be to derive the value from a path constant somewhere - I will leave it to those experienced with this project to develop a patch. This one-line hack was sufficient to get my Nextcloud working again though.

you can send this one line of your FIX? I would also like to use that until it is officially fixed

**Comment by plush** [comment]
I am sorry, I am not allowed to provide a patch directly for _reasons_. But it's easy to make the change yourself. But it's easy to make a patch yourself. The last argument to proc_open is an array of environment variables. It's empty by default. Just replace it with an array that sets `'LD_LIBRARY_PATH'` to `'.'`.

**Comment by plush** [comment]
The PHP manual [explains the array syntax and provides examples](https://www.php.net/manual/en/language.types.array.php). You can define the array inline, making it a true one-line change.

**Comment by TueTchen** [comment]
Thank you!
Now happy - works fine with the replacement of the empty array at the end of line 51: 
replace [ ] with ["LD_LIBRARY_PATH" => "."]

**Comment by plush** [comment]
Yep, that's all it takes.

**Comment by nilskamm** [comment]
That´s a really easy fix. Thanks a lot for sharing!!!

**Comment by alghanim-lab** [comment]
I tried your solution and I changed this line as follows:

![Bildschirmfoto 2022-03-30 um 16 56 09](https://user-images.githubusercontent.com/45328993/160865859-37c13da8-9259-488c-8cb4-a9e448c89728.png)

Unfortunately your solution doesn't work for me
in my system the variable LD_LIBRARY_PATH is not found with the command echo $LD_LIBRARY_PATH.

I used the find / -name command to search for and found the libgcc_s.so.1 object under /usr/lib/aarch64-linux-gnu/ and exported it but still doesn't work

does anyone have any ideas?

**Comment by dugwood** [comment]
@alghanim-lab you can change `$cmd` from:

`$cmd = './x2t ' . escapeshellarg($param);`

to:

`$cmd = 'export LD_LIBRARY_PATH=".";./x2t ' . escapeshellarg($param);`

Worked for me (but I have other issues...)

**Comment by eliotmasset** [comment]
Hi, all this solutions don't work for me and after a while, this work with : 
sudo cp apps/documentserver_community/3rdparty/onlyoffice/documentserver/server/FileConverter/bin/\*.so /usr/lib64/
and
sudo cp apps/documentserver_community/3rdparty/onlyoffice/documentserver/server/FileConverter/bin/\*.so.58 /usr/lib64/

Hope this can  help some people !

**Comment by plush** [comment]
The issue persists to this date. It's not stale/obsolete.

**Comment by chadek** [comment]
This issue should now be fix in v0.1.19
